### ISSUE TITLE
PHP Unit Tests don't work after updating to Laravel 11

### ISSUE THREAD

**Issue by ErwinLiemburg** [laravel/framework#50790]
### Laravel Version

11

### PHP Version

8.2

### Description

After updating to Laravel 11 I tried to run some PHP Unit tests that I used in the previous Laravel version (10). I use MySql as the main database and for testing I use Sqlite in memory (as set in phpunit.xml). Testing in Laravel 10 worked fine, but running the tests in Laravel 11 showed errors. 

Using MAMP 6.9 for PHP (8.2) and MySql (5.7.39) on Mac M1 with MacOS Sonoma 14.4.1.

### Steps To Reproduce

I updated a working Laravel 10 project to Laravel 11, following the steps in the Laravel documentation (Prologue > Upgrade guide). Dropped the vendor files, and reinstalled them using composer. At this point the application is still working as supposed to. But running a PHP Unit test

**Testresult:**
`php artisan test --filter AuthorisationTest

   FAIL  Tests\Feature\App\AuthorisationTest
  ⨯ admin can login
  ⨯ teacher can login
  ⨯ student can login
  ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────  
   FAILED  Tests\Feature\App\AuthorisationTest > admin can login                                                                        QueryException   
  SQLSTATE[HY000]: General error: 1 table sqlite_master may not be modified (Connection: sqlite, SQL: delete from sqlite_master where type in ('table', 'index', 'trigger'))

  at vendor/laravel/framework/src/Illuminate/Database/Connection.php:813
    809▕                     $this->getName(), $query, $this->prepareBindings($bindings), $e
    810▕                 );
    811▕             }
    812▕ 
  ➜ 813▕             throw new QueryException(
    814▕                 $this->getName(), $query, $this->prepareBindings($bindings), $e
    815▕             );
    816▕         }
    817▕     }

      +36 vendor frames 
  37  tests/Feature/App/AuthorisationTest.php:19`

Line 19 is the line of parent::setUp();

**AuthorisationTest.php:**
`<?php

namespace Tests\Feature\App;

use App\Enums\Roles;
use App\Models\User;
use App\Providers\RouteServiceProvider;
use Database\Seeders\PermissionsSeeder;
use Illuminate\Foundation\Testing\RefreshDatabase;
use Illuminate\Foundation\Testing\WithFaker;
use Tests\TestCase;

class AuthorisationTest extends TestCase
{
    use RefreshDatabase;

    protected function setUp(): void
    {
        parent::setUp();

        $this->seed(PermissionsSeeder::class);
    }

    public function test_admin_can_login(): void
    {

        $user = User::factory()->create();
        $user->assignRole(Roles::Admin->value);

        $response = $this->post('/login', [
            'email' => $user->email,
            'password' => 'password',
        ]);

        $this->assertAuthenticated();
        $response->assertRedirect(RouteServiceProvider::HOME);
    }

    public function test_teacher_can_login(): void
    {

        $user = User::factory()->create();
        $user->assignRole(Roles::Teacher->value);

        $response = $this->post('/login', [
            'email' => $user->email,
            'password' => 'password',
        ]);

        $this->assertAuthenticated();
        $response->assertRedirect(RouteServiceProvider::HOME);
    }

    public function test_student_can_login(): void
    {

        $user = User::factory()->create();
        $user->assignRole(Roles::Student->value);

        $response = $this->post('/login', [
            'email' => $user->email,
            'password' => 'password',
        ]);

        $this->assertAuthenticated();
        $response->assertRedirect(RouteServiceProvider::HOME);
        $response->assertStatus(302);
    }
}`

**phpunit.xml:**
`<?xml version="1.0" encoding="UTF-8"?>
<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/10.4/phpunit.xsd" bootstrap="vendor/autoload.php"
    colors="true">
    <testsuites>
        <testsuite name="Unit">
            <directory suffix="Test.php">./tests/Unit</directory>
        </testsuite>
        <testsuite name="Feature">
            <directory suffix="Test.php">./tests/Feature</directory>
        </testsuite>
    </testsuites>
    <coverage />
    <php>
        <env name="APP_ENV" value="testing" />
        <env name="BCRYPT_ROUNDS" value="4" />
        <env name="CACHE_DRIVER" value="array" />
        <env name="DB_CONNECTION" value="sqlite" />
        <env name="DB_DATABASE" value=":memory:" />
        <env name="MAIL_MAILER" value="array" />
        <env name="QUEUE_CONNECTION" value="sync" />
        <env name="SESSION_DRIVER" value="array" />
        <env name="TELESCOPE_ENABLED" value="false" />
    </php>
    <source>
    <include>
        <directory suffix=".php">./app</directory>
    </include>
    </source>
</phpunit>`

**Comment by ZacharyDuBois** [comment]
So, this looks like it is going to be some problem within your application code. Not Laravel. Laravel 11 is a breaking upgrade. Please ensure you follow the upgrade guide [here](https://laravel.com/docs/11.x/upgrade). There were a lot of DB related changes.

**Comment by ErwinLiemburg** [comment]
@ZacharyDuBois Yes, as mentioned I followed the upgrade guide. I will try to do some debugging to find out what is going on to see if it is in my application code.

**Comment by ErwinLiemburg** [comment]
Did some debugging. Looks like the issue occurd when the parent::SetUp() function is executed. Within that function it tries to refresh the test database. It tries to do a delete `delete from sqlite_master where type in ('table', 'index', 'trigger'))`, but for some reason it crashes. Looks like the sqlite_master table may not be modified.

Look like it's not from the application code, but Laravel code. It could be that I missed something in the upgrade guide, but I'm not sure what I missed.

<img width="672" alt="Screenshot 2024-03-27 at 19 13 46" src="https://github.com/laravel/framework/assets/55760701/1ef539cb-de7e-40ac-927f-bc3631f0c072">

**Comment by crynobone** [comment]
If you're resetting database between tests you should be using `DatabaseMigrations` trait instead of `RefreshDatabase`.
